app/auth/views/suppliers.py

```diff
@@ -1,28 +1,47 @@
-import pendulum
-from flask import jsonify, request
+from flask import jsonify
 from flask_login import current_user, login_required
 from app.auth import auth
 from app.utils import get_json_from_request
-from app.auth.suppliers import get_supplier, update_supplier_details, get_supplier_services
-from app.models import db, ServiceType, ServiceTypePrice, Supplier, Region, ServiceCategory
-from app.auth.helpers import role_required, abort, parse_date
-from itertools import groupby
-from app.swagger import swag
-from app.emails.prices import send_price_change_email
-from operator import itemgetter
+from app.auth.suppliers import get_supplier, update_supplier, get_supplier_services, get_all_suppliers
+from app.auth.helpers import role_required, is_current_supplier
 
 
-@auth.route('/supplier', methods=['GET'])
+@auth.route('/suppliers/<int:code>', methods=['GET'], endpoint='get_supplier')
 @login_required
-@role_required('supplier')
-def get_supplier_profile():
-    """Return the current supplier (role=supplier)
+@role_required('buyer', 'supplier')
+@is_current_supplier
```

Comment:
Won't this stop buyers from getting information about suppliers?

Reply:
it only stops suppliers who are not current_user - has no impact on buyers